shell/platform/android/io/flutter/embedding/engine/plugins/util/GeneratedPluginRegister.java

```diff
@@ -14,16 +14,57 @@
+   *
+   * <p>The Flutter tool generates that list in a class called {@code GeneratedPluginRegistrant}.
+   * That class contains generated code to register every plugin in the pubspec.yaml with a {@link
+   * FlutterEngine}. That file is generated in the Flutter project's directory.
+   *
+   * <p>The {@link FlutterEngine} (when a {@link FlutterEngine} is explicitly created) or the {@link
+   * io.flutter.embedding.android.FlutterActivity} (when a {@link FlutterEngine} is implicitly
+   * created) will automatically call the {@code GeneratedPluginRegistrant}, typically at the
+   * beginning of the application for normal full-Flutter apps, to register all plugins when a
+   * {@link FlutterEngine} is created.
+   *
+   * <p>Since the {@link FlutterEngine} belongs to the Flutter engine and the
+   * GeneratedPluginRegistrant class belongs to the app project, the {@link FlutterEngine} cannot
+   * place a compile-time dependency on GeneratedPluginRegistrant to invoke it. Instead, this class
+   * uses reflection to attempt to locate the generated file and then use it at runtime.
    *
    * <p>This method fizzles if the GeneratedPluginRegistrant cannot be found or invoked. This
    * situation should never occur, but if any eventuality comes up that prevents an app from using
    * this behavior, that app can still write code that explicitly registers plugins.
+   *
+   * <p>To disable this automatic plugin registration behavior:
+   *
+   * <ul>
+   *   <li>If you manually construct {@link FlutterEngine}s, construct the {@link FlutterEngine}
+   *       with the {@code automaticallyRegisterPlugins} construction parameter set to false.
+   *   <li>If you let the {@link io.flutter.embedding.android.FlutterActivity} or {@link
+   *       io.flutter.embedding.android.FlutterFragmentActivity} construct a {@link FlutterEngine}
+   *       implicitly, override the {@code configureFlutterEngine} implementation and don't call
+   *       through to the superclass implementation.
+   * </ul>
+   *
+   * <p>Disabling the automatic plugin registration or deferring it by calling this method
+   * explicitly may be useful in fine tuning the application launch latency characteristics for your
+   * application.
+   *
+   * <p>It's also possible to not use GeneratedPluginRegistrant and this method at all in order to
+   * fine tune not only when plugins are registered but which plugins are registered when.
+   * Inspecting the content of the GeneratedPluginRegistrant class will reveal that it's just going
+   * through each of the plugins referenced in pubspec.yaml and calling {@link
+   * io.flutter.embedding.engine.plugins.FlutterPlugin#onAttachedToEngine(io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding)}
+   * on each plugin. That code can be copy pasted and invoked directly per plugin to determine which
+   * plugin gets registered when in your own application's code. Note that when that's done without
```

Comment:
s/to determine which plugin gets registered when in/to determine when each plugin gets registered in

Reply:
there's a slight different. The point I was trying to make was that not all plugins need to be registered. If some code is circumstantially used, it could be done on conditions.